pkg/logql/log/parser.go

```diff
@@ -39,6 +40,8 @@ type JSONParser struct {
 	lbs *LabelsBuilder
 
 	keys internedStringSet
+
+	RequiredJsonLabels []string
```

Comment:
Q: So this `RequiredJsonLabels` are actually taken from the logql expression based on comes after `| json` filter correct?
 e.g: ` ... | json | foo="bar" `

Here the required labels are just `foo`? Is my understanding right?

Reply:
> 
Completely correct！
I added a test function to make the RequiredJsonLabels logic easier to understand.

![image](https://user-images.githubusercontent.com/9583245/154447140-3d6ab971-3ff4-41b5-a829-67021d6edecf.png)